src/TraceEvent/Computers/StartStopActivityComputer.cs

```diff
@@ -1140,7 +1140,8 @@ void FixAndProcessWindowsASP(TraceEvent data, KeyValuePair<Guid, Guid>[] threadT
                 if (creator == null)        // If we don't have a creator for the ASP.NET event, make one since WCF and others are children of it and we want to see that 
                 {
                     // We create another 
-                    creator = OnStart(data, extraStartInfo, &relatedActivityId, thread, null, "RecHttp", useCurrentActivityForCreatorAsFallback: false);
+                    // It is closer to Start Event than the Send Event, uses &activityID.
+                    creator = OnStart(data, extraStartInfo, &activityID, thread, null, "RecHttp", useCurrentActivityForCreatorAsFallback: false);
```

Comment:
I am taking your word that this change makes sense (that for these ASP.NET events using the activityID instead of the relatedActivity is the right thing to do).